smtp/client.js

```diff
@@ -295,7 +295,15 @@ class Client {
 	_sendmessage(stack) {
 		const stream = stack.message.stream();
 
-		stream.on('data', data => this.smtp.message(data));
+		stream.on('data', (data) => {
+			if(this.smtp.state() === state.NOTCONNECTED) {
```

Comment:
if we stored the stream in a non-local way, we could attach a listener on the SMTP connection directly and close the stream immediately upon SMTP timeout / disconnect.

what are your thoughts on doing that there instead of here?

Reply:
Ideally yeah, we should close the stream as soon as we know the original socket is dead. I was trying to introduce minimal change for easy review/integration since we are facing this issue in live code. If you are okay with it, I can take a crack at closing the stream when socket dies.

In all the scenarios except timeouts `SMTP.close` call is followed by `caller(callback, ...)` call with proper error. It should be trivial to introduce a call to `caller(callback, ...)` with error upon timeout. Client.js can then close the stream in `_senddone` if required. 

There are two potential issues:
1. For Secure SMTP there are two timeout monitors started. Second one is unreferenced, and can't be cancelled in current setup. From what I understood, the second one (in starttls) could simply replace `this.monitor` (after stopping the previous one). Is there some other reason for having the second monitor which I missed?

2. The message stream when closed with a parameter fires an 'error' event. The error event handler triggers a close (again) and _senddone. That cycle would need to be handled specially.

Reply:
nice break down. if you are facing this in live code right now, i'ld use a local branch/fork since your proposed fix is likely "good enough".

i'ld be open to you taking a crack at it. `monitor` takes in the `socket` as a parameter, so i think the intent is to have a unified place to handle timeouts. if there is double handling, perhaps this is an opportunity to simplify it.